### DISCUSSION TITLE
setValue does not trigger re-render of registered parent object

### DISCUSSION THREAD

**Discussion by matt-tingen** [react-hook-form/react-hook-form#6538] [Q&A]
https://codesandbox.io/s/hungry-brown-wcm6b?file=/src/App.tsx

In the sandbox, a `Controller` registers the object field `user`. When using `setValue` to set a property on `user`, the notation used determines whether the `Controller` re-renders or not:
- `setValue('user', { name: '...' })` causes a re-render
- `setValue('user.name', '...')` does not cause a re-render

Both calls update the internal value as can be seen when submitting the form.

Steps to repro:
- Click `Set "user.name"` button

Expected:
- "No value" text is replaced with "set as user.name"

Observed: 
- "No value" remains

**Comment by matt-tingen** [comment]
@bluebill1049 Apologies for the bump, I think this may have slipped by. Would you mind taking a look when you have some time? It seems like a bug to me that the two usages of `setValue` have different behavior, but I'm not sure.

**Reply by bluebill1049** [reply]
i think we may be able to improve on this one, i will convert this into an issue.

**Comment by bluebill1049** [comment]
after a quick investigation, the reason we couldn't re-render was that no such input could be found.  

- `user` could be found as get registered and we would be able to update the value and render
- `user.name` there is no such input get found and hence failed.
